meson.build

```diff
@@ -211,17 +195,18 @@ endif
 
 # windows stuff
 if is_windows
-	devel_args += [
-		'-D_ITERATOR_DEBUG_LEVEL=0',
-		'-D_WINSOCK_DEPRECATED_NO_WARNINGS',
-		'-D_SCL_SECURE_NO_WARNINGS',
-		'-D_CRT_SECURE_NO_WARNINGS',
-	]
-	devel_args += has_exceptions ? '-D_HAS_EXCEPTIONS=1' : '-D_HAS_EXCEPTIONS=0'
+	universal_args += has_exceptions ? '-D_HAS_EXCEPTIONS=1' : '-D_HAS_EXCEPTIONS=0'
 elif is_release
 	overrides += 'strip=true'
 endif
 
+# library is being compiled as a separate lib target
+if get_option('compile_library')
+	universal_args += '-DTOML_HEADER_ONLY=0'
+	if get_option('default_library') != 'static'
+		universal_args += '-DTOML_SHARED_LIB=1'
+	endif
+endif
```

Comment:
These shouldn't be needed, because you only need to use them if you're using the compiled library. So they should be set as `compile_args` of the `tomlplusplus_dep` object, so that every consumer (in our case the tests and examples) uses them